pkg/controller/controller.go

```diff
@@ -471,10 +463,21 @@ func (c *Controller) runAction(ctx context.Context, t *tomb.Tomb, as *crv1alpha1
 			var msg string
 			if err == nil {
 				c.updateActionSetRunningPhase(ctx, as, p.Name())
+				progressTrackCtx, doneProgressTrack := context.WithCancel(ctx)
+				defer doneProgressTrack()
+				go func() {
+					// progress update is computed on a best-effort basis.
+					// if it exits with error, we will just log it.
+					if err := progress.TrackActionsProgress(progressTrackCtx, c.crClient, as.GetName(), as.GetNamespace(), p); err != nil {
```

Comment:
`TrackActionsProgress` continuously updates the current progress in actionset phase right?
Should we rename it to something that is more clearly conveying that, for example `IterateActionProgress`? Basically a word that conveys something is being improved continuously. :smile:

Reply:
I can settle with `UpdateActionsetsProgress`